plugins/audio-s5gen2/fu-audio-s5gen2-ble-device.c

```diff
@@ -0,0 +1,615 @@
+		switch (buf[i]) {
+		case FU_QC_GAIA_V3_FEATURES_CORE:
+			self->feature.core = buf[i + 1];
+			g_debug("Core feature version: %u", self->feature.core);
+			break;
+		case FU_QC_GAIA_V3_FEATURES_DFU:
+			self->feature.dfu = buf[i + 1];
+			g_debug("DFU feature version: %u", self->feature.dfu);
+			break;
+		default:
+			break;
+		}
+	}
+
+	/* request the rest of the list */
+	if (fu_struct_qc_gaia_v3_supported_features_get_more_features(resp) == FU_QC_MORE_MORE)
+		return fu_qc_s5gen2_ble_device_get_features(self, TRUE, error);
+
+	return TRUE;
+}
+
+static gboolean
+fu_qc_s5gen2_ble_device_get_serial(FuQcS5gen2BleDevice *self, GError **error)
+{
+	guint8 buf[FU_QC_S5GEN2_BLE_DEVICE_BUFFER_SZ] = {0};
+	gsize read_len;
+	g_autoptr(GByteArray) req = fu_struct_qc_gaia_v3_serial_req_new();
+	g_autoptr(GByteArray) validate = NULL;
+	g_autofree gchar *serial = NULL;
+
+	fu_struct_qc_gaia_v3_serial_req_set_vendor_id(req, self->vid_v3);
+
+	if (!fu_qc_s5gen2_ble_device_send(self, req->data, req->len, error))
+		return FALSE;
+
+	if (!fu_qc_s5gen2_ble_device_recv(self, buf, sizeof(buf), &read_len, error))
+		return FALSE;
+
+	/* Check if response is valid */
+	validate =
+	    fu_struct_qc_gaia_v3_serial_parse(buf, FU_STRUCT_QC_GAIA_V3_SERIAL_SIZE, 0, error);
+	if (validate == NULL)
+		return FALSE;
+
+	serial = g_strndup((gchar *)(buf + FU_STRUCT_QC_GAIA_V3_SERIAL_SIZE),
+			   read_len - FU_STRUCT_QC_GAIA_V3_SERIAL_SIZE);
+
+	fu_device_set_serial(FU_DEVICE(self), serial);
+
+	return TRUE;
+}
+
+static gboolean
+fu_qc_s5gen2_ble_device_get_variant(FuQcS5gen2BleDevice *self, GError **error)
+{
+	guint8 buf[FU_QC_S5GEN2_BLE_DEVICE_BUFFER_SZ] = {0};
+	gsize read_len;
+	g_autoptr(GByteArray) req = fu_struct_qc_gaia_v3_variant_req_new();
+	g_autoptr(GByteArray) validate = NULL;
+	g_autofree gchar *variant = NULL;
+
+	fu_struct_qc_gaia_v3_variant_req_set_vendor_id(req, self->vid_v3);
+
+	if (!fu_qc_s5gen2_ble_device_send(self, req->data, req->len, error))
+		return FALSE;
+
+	if (!fu_qc_s5gen2_ble_device_recv(self, buf, sizeof(buf), &read_len, error))
+		return FALSE;
+
+	/* Check if response is valid */
+	validate =
+	    fu_struct_qc_gaia_v3_variant_parse(buf, FU_STRUCT_QC_GAIA_V3_VARIANT_SIZE, 0, error);
+	if (validate == NULL)
+		return FALSE;
+
+	variant = g_strndup((gchar *)(buf + FU_STRUCT_QC_GAIA_V3_VARIANT_SIZE),
+			    read_len - FU_STRUCT_QC_GAIA_V3_VARIANT_SIZE);
+
+	/* create the GUID based on variant read from device */
+	fu_device_add_instance_str(FU_DEVICE(self), "GAIA", variant);
+	fu_device_build_instance_id_full(FU_DEVICE(self),
+					 FU_DEVICE_INSTANCE_FLAG_VISIBLE |
+					     FU_DEVICE_INSTANCE_FLAG_QUIRKS,
+					 NULL,
+					 "BLUETOOTH",
+					 "GAIA",
+					 NULL);
+
+	return TRUE;
+}
+
+static gboolean
+fu_qc_s5gen2_ble_device_register_notification(FuQcS5gen2BleDevice *self, GError **error)
+{
+	guint8 buf[FU_QC_S5GEN2_BLE_DEVICE_BUFFER_SZ] = {0};
+	gsize read_len;
```

Comment:
i think coverity would prefer you to have `read_len = 0` as it's passed by ref in fu_qc_s5gen2_ble_device_recv and it doesn't technically have to have an assigned value.

Reply:
`read_len = 0` also means there's one less coverity warning to ignore :)